src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphSubmissionDAO.scala

```diff
@@ -66,12 +79,19 @@ class GraphSubmissionDAO extends SubmissionDAO with GraphDAO {
   /** create a submission (and its workflows) */
   override def save(workspaceNamespace: String, workspaceName: String, submission: Submission, txn: RawlsTransaction) =
     txn withGraph { db =>
-      val workspace = getWorkspaceVertex(db, workspaceNamespace, workspaceName).getOrElse(throw new IllegalArgumentException(s"workspace ${workspaceNamespace}/${workspaceName} does not exist"))
+      val workspaceVertex = getWorkspaceVertex(db, workspaceNamespace, workspaceName).getOrElse(throw new IllegalArgumentException(s"workspace ${workspaceNamespace}/${workspaceName} does not exist"))
       val submissionVertex = setVertexProperties(submission, addVertex(db, null))
-      workspace.addEdge(ExecutionEdgeTypes.submissionEdgeType, submissionVertex)
+      val submissionEntityVertex =  getSinglePipelineResult(new GremlinPipeline(workspaceVertex).out(submission.entityType).filter(hasProperty("_name", submission.entityName))).
+        getOrElse(throw new IllegalArgumentException(s"entity of type ${submission.entityType} named ${submission.entityName} does not exist in workspace ${workspaceNamespace}/${workspaceName}"))
+      submissionVertex.addEdge(ExecutionEdgeTypes.runOnEdgeType, submissionEntityVertex)
+      workspaceVertex.addEdge(ExecutionEdgeTypes.submissionEdgeType, submissionVertex)
+
       submission.workflow.foreach { workflow =>
         val workflowVertex = setVertexProperties(workflow, addVertex(db, null))
-        workspace.addEdge(ExecutionEdgeTypes.workflowEdgeType, workflowVertex)
+        val entityVertex =  getSinglePipelineResult(new GremlinPipeline(workspaceVertex).out(workflow.entityType).filter(hasProperty("_name", workflow.entityName))).
+                               getOrElse(throw new IllegalArgumentException(s"entity of type ${workflow.entityType} named ${workflow.entityName} does not exist in workspace ${workspaceNamespace}/${workspaceName}"))
+        workflowVertex.addEdge(ExecutionEdgeTypes.runOnEdgeType, entityVertex)
```

Comment:
So the edge goes from workflow to entity, not entity to workflow, we would just need to follow in going edges from entities to find associated workflows I guess